src/api/app/controllers/webui/users/notifications_controller.rb

```diff
@@ -42,21 +41,33 @@ def update
   private
 
   def set_filter_type
-    @filter_type = params[:kind] || 'all'
-    raise FilterNotSupportedError if @filter_type.present? && ALLOWED_FILTERS.exclude?(@filter_type)
+    @filter_type = params[:kind].presence || ['all']
+    @filter_type = [@filter_type] if @filter_type.instance_of?(String) # in case just one value, we want an array anyway
+    raise FilterNotSupportedError unless (@filter_type - ALLOWED_FILTERS).empty?
+
+    @filter_type
   end
 
   def set_filter_state
-    @filter_state = params[:state] || 'unread'
-    raise FilterNotSupportedError if @filter_state.present? && ALLOWED_STATES.exclude?(@filter_state)
+    @filter_state = params[:state].presence || ['unread']
+    @filter_state = [@filter_state] if @filter_state.instance_of?(String) # in case just one value, we want an array anyway
+    raise FilterNotSupportedError unless (@filter_state - ALLOWED_STATES).empty?
+  end
+
+  def set_filter_project
+    @filter_project = params[:project] || []
+  end
+
+  def set_filter_group
+    @filter_group = params[:group] || []
   end
 
   def set_notifications
     @notifications = User.session!.notifications.for_web.includes(notifiable: [{ commentable: [{ comments: :user }, :project, :bs_request_actions] }, :bs_request_actions, :reviews])
-    @notifications = @notifications.for_project_name(params[:project]) if params[:project].present?
-    @notifications = @notifications.for_group_title(params[:group]) if params[:group].present?
-    @notifications = filter_notifications_by_type(@notifications, @filter_type)
+    @notifications = filter_notifications_by_project(@notifications, @filter_project)
+    @notifications = filter_notifications_by_group(@notifications, @filter_group)
     @notifications = filter_notifications_by_state(@notifications, @filter_state)
+    @notifications = filter_notifications_by_type(@notifications, @filter_type)
```

Comment:
I think `filter_notifications_by_type` and `@filter_type` should be renamed to `*_kind` after the renaming from `type` to `kind`. Something we can do later as well.

Reply:
It was a straight forward renaming, fixed now. Thank you